model/project_ref_test.go

```diff
@@ -2304,24 +2304,89 @@ func TestGetProjectTasksWithOptions(t *testing.T) {
 
 func TestUpdateNextPeriodicBuild(t *testing.T) {
 	assert := assert.New(t)
-	assert.NoError(db.Clear(ProjectRefCollection))
 	now := time.Now().Truncate(time.Second)
-	p := ProjectRef{
-		Id: "proj",
-		PeriodicBuilds: []PeriodicBuildDefinition{
-			{ID: "1", NextRunTime: now},
-			{ID: "2", NextRunTime: now.Add(1 * time.Hour)},
+	later := now.Add(1 * time.Hour)
+	muchLater := now.Add(10 * time.Hour)
+	for name, test := range map[string]func(*testing.T){
+		"updatesProject": func(t *testing.T) {
+			p := ProjectRef{
+				Id: "proj",
+				PeriodicBuilds: []PeriodicBuildDefinition{
+					{ID: "1", NextRunTime: now},
+					{ID: "2", NextRunTime: later},
+				},
+				RepoRefId: "repo",
+			}
+			repoRef := RepoRef{ProjectRef{
+				Id: "repo",
+				PeriodicBuilds: []PeriodicBuildDefinition{
+					{ID: "2", NextRunTime: later},
+				},
+			}}
+			assert.NoError(p.Insert())
+			assert.NoError(repoRef.Upsert())
+
+			assert.NoError(UpdateNextPeriodicBuild("proj", "2", muchLater))
+			dbProject, err := FindBranchProjectRef(p.Id)
+			assert.NoError(err)
+			assert.NotNil(dbProject)
```

Comment:
I'm fine to change it but I'm slightly against making this a requirement for tests, since we're pretty inconsistent about this throughout the codebase (and we aren't checking if lengths are correct or etc so there's still room for panics), and it can make tests longer/verbose and slow up code reviews with nit type stuff.

Reply:
I don't have strong opinions either way, it's just a minor inconvenience when a test panics and stop all the other tests in the package from proceeding.